src/Validation.PackageSigning.ValidateCertificate/OnlineCertificateVerifier.cs

```
@@ -0,0 +1,188 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using NuGet.Services.Validation;
+
+namespace Validation.PackageSigning.ValidateCertificate
+{
+    /// <summary>
+    /// Performs online revocation verification for a <see cref="X509Certificate2"/>.
+    /// </summary>
+    public class OnlineCertificateVerifier : ICertificateVerifier
+    {
+        public const string CodeSigningCertificateOid = "1.3.6.1.5.5.7.3.3";
+
+        /// <summary>
+        /// If building the chain for a certificate results in one of these flags, the certificate will be
+        /// marked as <see cref="EndCertificateStatus.Invalid"/>. Notable missing flags include:
+        ///
+        /// * NotTimeValid - Expired certificates *may* be valid during revalidations (but NOT during initial push validation).
+        /// * Revoked - Revoked certificates are marked as <see cref="EndCertificateStatus.Revoked"/>.
+        /// * OfflineRevocation - Certificates whose online revocation fail are marked as <see cref="EndCertificateStatus.Unknown"/>.
+        /// </summary>
+        public const X509ChainStatusFlags InvalidCertificateFlags =
```

This should take all flags turned on and subtract special cases.

---

Talked to @joelverhagen offline. We should be explicit as flags that are added in the future do not necessarily invalidate a package (`EndCertificateStatus.Unknown` or `EndCertificateStatus.Revoked` may be appropriate instead). Instead, the code should fail to validate a package whose chain status is unknown (see [this line](https://github.com/NuGet/NuGet.Jobs/pull/309/files#diff-64f8ba384ceda94fc9cc25255ba92da5R83)).

I will add a unit test that enforces that all `X509ChainStatusFlags` are either listed in `InvalidCertificateFlags` or known special cases.

---

Added unit test.